### PR TITLE
ENH: HeSpaDDA updated the distribution of processors independent on t…

### PR DESCRIPTION
**Pull request by govarguz**
ENH: HeSpaDDA updated the distribution of processors independent on t…he axis sequence X,Y,Z or Z,Y, X; among other possible combiataion  [upgrade to HeSpaDDA v1.1]

### CODE REVIEW

**Comment by junghans** [comment]
Has a conflict.

**Comment by junghans** [comment]
Also https://chris.beams.io/posts/git-commit/: _Limit the subject line to 50 characters_!

**Comment by govarguz** [comment]
@junghans conflict solved, there was an issue with the BoxSize since it is a list and the control sentence <isinstance> was not working I move it to another real variable _HeSpaDDA_ uses. BTW thnks for the commit message heads up!
Oh a test is not really needed, because this enhancement is only an extension to _HeSpaDDA_ whenever the box is let's say "bigger" in Y or Z dimensions. For example,
[20,40,20] and [20,20,40] will now allocate processors as if it would be [40,20,20] (included before in _HeSpaDDA_.

**Comment by junghans** [comment]
GitHub still has a conflict, you need to do a:
```
git remote update
git merge origin/master
```

**Comment by junghans** [comment]
> Oh a test is not really needed, because this enhancement is only an extension to _HeSpaDDA_ whenever the box is let's say "bigger" in Y or Z dimensions. For example,
> [20,40,20] and [20,20,40] will now allocate processors as if it would be [40,20,20] (included before in _HeSpaDDA_.
I would still like a test in testsuite.

**Comment by govarguz** [comment]
This PR is the first of a couple that I have pending about the whole HeSpaDDA algorithms. In this case it tackles the nodeGrids in different schemes as can be seen in the Testsuite I am also adding. Cheers!

File: testsuite/FixedLocalTuple/TestFixedLocalTupleList.py [hunk 1]
```diff
@@ -54,7 +57,7 @@ def setUp(self) :
         cellGrid       = espressopp.tools.decomp.cellGrid(box, nodeGrid, cutoff, system.skin)
         
         print 'NodeGrid = %s'%(nodeGrid,)
-        print 'CellGrid = %s'%cellGrid
+        print 'CellGrid = %s'%(cellGrid,)
```
Comment (jkrajniak) [inline testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:60]: Why?

File: testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py [hunk 2]
```diff
@@ -29,13 +32,16 @@
 class TestCaseLangevinThermostatOnGroup(unittest.TestCase):
     def setUp(self):
         system = espressopp.System()
-        system.bc = espressopp.bc.OrthorhombicBC(system.rng, (10, 10, 10))
+	box=(10,10,10)
```
Comment (jkrajniak) [inline testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py:35]: The indentation seems to be incorrect

File: testsuite/minimize_energy/test_minimize_energy.py [hunk 3]
```diff
@@ -33,10 +36,14 @@ class TestMinimizeEnergy(unittest.TestCase):
     def setUp(self):
 
         system = espressopp.System()
-        system.bc = espressopp.bc.OrthorhombicBC(system.rng, (10, 10, 10))
+	box=(10,10,10)
```
Comment (jkrajniak) [inline testsuite/minimize_energy/test_minimize_energy.py:39]: indentation?

File: testsuite/FixedLocalTuple/TestFixedLocalTupleList.py (hunk 1 above)
Comment (govarguz) [reply testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:60]: Seems to be a bug with Int3D(), but now it is solved

File: testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py (hunk 2 above)
Comment (govarguz) [reply testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py:35]: True

File: testsuite/minimize_energy/test_minimize_energy.py (hunk 3 above)
Comment (govarguz) [reply testsuite/minimize_energy/test_minimize_energy.py:39]: Done

File: src/tools/decomp.py [hunk 4]
```diff
@@ -397,10 +511,10 @@ def tuneSkin(system, integrator, minSkin=0.01, maxSkin=1.5, precision=0.001, pri
 
     fi = (1.0 + math.sqrt(5.0)) / 2.0  # golden ratio
 
-    npart = espressopp.analysis.NPart(system).compute()
+    npart = 100 #espressopp.analysis.NPart(system).compute()
```
Comment (junghans) [inline src/tools/decomp.py:514]: Hmm, is the particle number always 100?

File: src/tools/decomp.py [hunk 5]
```diff
@@ -397,10 +511,10 @@ def tuneSkin(system, integrator, minSkin=0.01, maxSkin=1.5, precision=0.001, pri
...
 
     # this is an empirical formula in order to get the appropriate number of steps
-    nsteps = int(espressopp.MPI.COMM_WORLD.size * 1000000.0 / float(npart))
+    nsteps = 10 #int(espressopp.MPI.COMM_WORLD.size * 1000000.0 / float(npart))
```
Comment (junghans) [inline src/tools/decomp.py:517]: Always 10?

File: src/tools/decomp.py [hunk 6]
```diff
@@ -449,9 +563,9 @@ def tuneSkin(system, integrator, minSkin=0.01, maxSkin=1.5, precision=0.001, pri
 
 
 def printTimeVsSkin(system, integrator, minSkin=0.01, maxSkin=1.5, skinStep=0.005):
-    npart = espressopp.analysis.NPart(system).compute()
+    npart = 100 # espressopp.analysis.NPart(system).compute()
```
Comment (junghans) [inline src/tools/decomp.py:566]: Same question.

File: src/tools/decomp.py [hunk 7]
```diff
@@ -449,9 +563,9 @@ def tuneSkin(system, integrator, minSkin=0.01, maxSkin=1.5, precision=0.001, pri
...
     # this is an empirical formula in order to get the appropriate number of steps
-    nsteps = int(espressopp.MPI.COMM_WORLD.size * 10000000.0 / float(npart))
+    nsteps = 10 # int(espressopp.MPI.COMM_WORLD.size * 10000000.0 / float(npart))
```
Comment (junghans) [inline src/tools/decomp.py:568]: Same.

File: src/tools/loadbal.py [hunk 8]
```diff
@@ -1,4 +1,5 @@
-#  Copyright (C) 2017,2018(1H)
+#  Copyright (C) 2012,2013,2017,2018,2020(1H)
+#      Jozef Stefan Institute
```
Comment (junghans) [inline src/tools/loadbal.py:2]: I think you need a new line as `Jozef Stefan Institute` was involved in  pre 2020.

File: src/tools/loadbal.py [hunk 9]
```diff
@@ -1,4 +1,5 @@
-#  Copyright (C) 2017,2018(1H)
+#  Copyright (C) 2012,2013,2017,2018,2020(1H)
```
Comment (junghans) [inline src/tools/loadbal.py:1]: 2012 got added?

File: testsuite/FixedLocalTuple/TestFixedLocalTupleList.py [hunk 10]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env python2
 #
+#  Copyright (C) 2020(Dr. Horacio V Guzman)
```
Comment (junghans) [inline testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:3]: Whatever you do just make it consistent, use full name or `(H)` or `(1H)`.

File: testsuite/FixedLocalTuple/TestFixedLocalTupleList.py (hunk 1 above)
Comment (junghans) [reply testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:60]: So still why?

File: testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py [hunk 11]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env python2
 #
+#  Copyright (C) 2020(Dr. Horacio V Guzman)
```
Comment (junghans) [inline testsuite/langevin_thermostat_on_group/langevin_thermostat_on_group.py:3]: Same, use full name everywhere or just `(H)`.

**Comment by govarguz** [comment]
I have checked all (c) related to this PR, regarding the commented variables in tuneSkin I have uncommented them. Other answers can be found above!

File: testsuite/FixedLocalTuple/TestFixedLocalTupleList.py (hunk 1 above)
Comment (govarguz) [reply testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:60]: Well, the %s denotes a string like (1,2,3) which is separated by comma. the original problem was that the cellGrid was in some testCases returned as string and not as an Int3D. Nonentheless, for either case "pure string (1,2,3)" or "Int3D(1,2,3)" the current print command is valid.

Comment (junghans) [reply testsuite/FixedLocalTuple/TestFixedLocalTupleList.py:60]: ok

File: src/tools/decomp.py [hunk 12]
```diff
@@ -245,6 +355,7 @@ def nodeGridSimple(n):
                     d2 = j
                     d3 = i
                     ijkmax = i * i + j * j + k * k
+    print d1, d2, d3
```
Comment (junghans) [inline src/tools/decomp.py:358]: Is this an extra arbitrary print?

Comment (govarguz) [reply src/tools/decomp.py:358]: yes, I added it to know that HeSpaDDA is not used in this cases. Morevoer, I should better explicitly write that when the nodeGridSImple(n) is used HeSpaDDA is not doing anything. Thanks!

Comment (junghans) [reply src/tools/decomp.py:358]: Add some prefix string, I really hate it when 3 numbers randomly show up on my screen without an idea from where.

Comment (govarguz) [reply src/tools/decomp.py]: done!

**Pull request merged**